src/seb/interfaces/model.py

```diff
@@ -131,29 +130,22 @@ def encode(
         """
         return self.model.encode(sentences, batch_size=batch_size, task=task, **kwargs)
 
-    def encode_queries(self, queries: list[str], batch_size: int, **kwargs):
+    def encode_queries(self, queries: list[str], batch_size: int, **kwargs):  # noqa
         try:
-            return self.model.encode_queries(queries, batch_size=batch_size, **kwargs)
+            return self.model.encode_queries(queries, batch_size=batch_size, **kwargs)  # type: ignore
         except AttributeError:
             return self.encode(queries, task=None, batch_size=batch_size, **kwargs)
 
-    def encode_corpus(self, corpus: list[dict[str, str]], batch_size: int, **kwargs):
+    def encode_corpus(self, corpus: list[dict[str, str]], batch_size: int, **kwargs):  # noqa
         try:
-            return self.model.encode_corpus(corpus, batch_size=batch_size, **kwargs)
+            return self.model.encode_corpus(corpus, batch_size=batch_size, **kwargs)  # type: ignore
         except AttributeError:
             sep = " "
-            if type(corpus) is dict:
+            if isinstance(corpus, dict):
```

Comment:
@x-tabdeveloping - seemed like this was a consistent pattern - just wanted to let you know

Reply:
So you mean we should always use `isinstance`? I tend to do that so I agree